manager/integration/tests/test_basic.py

```diff
@@ -1891,17 +1892,89 @@ def test_pvc_storage_class_name_from_backup_volume(): # NOQA
 
     Then
     - the backupvolume's status.storageClassName should be
-      longhorn-sc-name-recorded
+      longhorn-test
 
     When
     - Restore the backup to a new volume
     - Create PV/PVC from the new volume with create new PVC option
 
     Then
-    - The new PVC's storageClassName should still be longhorn-sc-name-recorded
+    - The new PVC's storageClassName should still be longhorn-test
     - Verify the restored data is the same as original one
     """
-    pass
+    volume_size = str(300 * Mi)
+    create_storage_class(storage_class)
+
+    pod_name = "pod-" + pvc_name
+    pvc['metadata']['name'] = pvc_name
+    pvc['spec']['storageClassName'] = storage_class['metadata']['name']
+    pvc['spec']['resources']['requests']['storage'] = volume_size
+    common.create_pvc(pvc)
+
+    pv = common.wait_and_get_pv_for_pvc(core_api, pvc_name)
+    assert pv.status.phase == "Bound"
+
+    test_pod = pod_make(pod_name)
+    test_pod['metadata']['name'] = pod_name
+    test_pod['spec']['volumes'] = [{
+        'name': test_pod['spec']['containers'][0]['volumeMounts'][0]['name'],
+        'persistentVolumeClaim': {'claimName': pvc_name},
+    }]
+    create_and_wait_pod(core_api, test_pod)
+
+    test_data = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_name, test_data)
+
+    volume_name = pv.spec.csi.volume_handle
+    volume_id = client.by_id_volume(volume_name)
+    snapshot = volume_id.snapshotCreate()
+
+    volume_id.snapshotBackup(name=snapshot.name)
+    wait_for_backup_completion(client, volume_name, snapshot.name)
+
+    # in nfs backupstore, bv.storageClassName sometimes were empty
+    # due to timing issue
+    for i in range(RETRY_COMMAND_COUNT):
+        bv, b = find_backup(client, volume_name, snapshot.name)
+        if bv.storageClassName != "":
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert bv.storageClassName == storage_class['metadata']['name']
```

Comment:
Just to confirm, according to the test skeleton `status.storageClassName` is asserted. Do we have two fields `StorageClassName` and `status.storageClassName`?

Reply:
Checked and there are only `status.storageClassName` in `backupvolumes.longhorn.io`
```bash
> k get backupvolume pvc-d8869f46-c205-40c3-851e-85c8f37c1410 -n longhorn-system -o yaml | grep storageClassName
  storageClassName: longhorn-test
```